src/NzbDrone.Core/Music/TrackService.cs

```diff
@@ -88,11 +88,19 @@ public Track FindTrackByTitle(int artistId, int albumId, int mediumNumber, strin
                     Position = normalizedReleaseTitle.IndexOf(Parser.Parser.NormalizeEpisodeTitle(track.Title), StringComparison.CurrentCultureIgnoreCase),
                     Length = Parser.Parser.NormalizeEpisodeTitle(track.Title).Length,
                     Track = track
-                })
-                                .Where(e => e.Track.Title.Length > 0 && e.Position >= 0)
-                                .OrderBy(e => e.Position)
-                                .ThenByDescending(e => e.Length)
-                                .ToList();
+                });
+
+            if (trackNumber == 0)
```

Comment:
Should this be if = 0 or null? Can trackNumber be null?

Reply:
Just did a look the value mapping to `TrackNumber` in the API it set to not null in mb db.

Reply:
Ok, so this logic should be good?